File: java_solutions/367.valid-perfect-square.java
/*
 * @lc app=leetcode id=367 lang=java
 *
 * [367] Valid Perfect Square
 */

class Solution {
    public boolean isPerfectSquare(int num) {
        return (int)Math.pow(num,0.5) == Math.pow(num,0.5);
    }
}

File: kotlin_solutions/367.valid-perfect-square.kt
/*
 * @lc app=leetcode id=367 lang=kotlin
 *
 * [367] Valid Perfect Square
 */

class Solution {fun isPerfectSquare(num: Int): Boolean = generateSequence(1L) { it + 1 }.first { it * it >= num }.let { it * it == num.toLong() }}
